distributed/tests/test_resources.py

```diff
@@ -393,3 +393,15 @@ def g(dask_worker):
     logs = client.run(g)
     assert logs[a["address"]]
     assert not logs[b["address"]]
+
+
+@gen_cluster(
+    client=True, nthreads=[("127.0.0.1", 2, {"resources": {"A": 1}}),],
+)
+def test_resources_do_not_block_all_threads(c, s, a):
+    constrained = c.submit(slowinc, 1, delay=0.2, resources={"A": 1})
+    unconstrained = c.submit(inc, 2)
+
+    # Fast, unconstrained work finished while slower, constrained is processing.
+    yield wait(unconstrained)
+    assert s.get_task_status(keys=[constrained.key]) == {constrained.key: "processing"}
```

Comment:
We tend to discourage the use of internal API in tests when possible.  That way we're more easily able to change internals without modifying very many tests.  Maybe you wanted to test that `constrained.status == "pending"` or something similar?  

This test is also a little bit prone to intermittent failure.  The CI systems can be sluggish sometimes, and I wouldn't be surprised to see a 200ms delay caused by something like GC, which would occasionally cause this test to fail.  I'm not sure how exactly to solve the problem.  Increasing the delay to something like 500ms might be safer?

Reply:
I agree with both points.  I didn't realize I was using an internal API.  (I simply was cribbing from what was done elsewhere in this file.)

I'm not sure I want "pending" here.  I think(?) I want to verify that it is still running.  (And, in fact, I should verify this before I start the fast tasks.)

My preference would be to examine details about how the tasks were executed -- essentially, I want to verify the tasks ran side-by-side.  Is there a way to run the tasks and verify after-the-fact what worker/thread each task ran on and it's start/end time?